src/controllers/EthereumBlockchainController.ts

```diff
@@ -461,31 +470,79 @@ export class EthereumBlockchainController extends AbstractBlockchainController {
 		return null;
 	}
 
-	async getTokenAttachments(msg: IEVMMessage): Promise<YlideTokenAttachment | null> {
+	async getSupplement(msg: IEVMMessage): Promise<Supplement | null> {
```

Comment:
Propose to have general method which can be called for any IEVMMessage. If there will be some supplement information (from YlidePay or YlideSafe) it will be retrieved, otherwise return null.